core/invoker/src/main/scala/whisk/core/containerpool/DockerContainer.scala

```diff
@@ -162,6 +162,9 @@ class DockerContainer(id: ContainerId, ip: ContainerIp)(
                 transid.finished(this, start.copy(start = r.interval.start), s"initialization result: ${r.toBriefString}", endTime = r.interval.end)
             case Failure(t) =>
                 transid.failed(this, start, s"initializiation failed with $t")
+        }.recoverWith {
+            case t =>
+                Future.failed(InitializationError(ActivationResponse.whiskError("action failed to initialize"), Interval(Instant.EPOCH, Instant.EPOCH)))
```

Comment:
can you add a comment that explain which class of errors will reach here? 
in other words it is not the case that all init failures are whisk errors (in Success(r) above, you capture failures to connect to the container as well?).